src/renderer/runtime/engine.store.ts

```diff
@@ -157,17 +159,20 @@ function createWriteBuffer() {
         });
 
         do {
-          const { id } = await sendDataToGrid(incoming.descr);
-          if (
-            incoming.responseRequired &&
-            incoming.filter !== undefined &&
-            incoming.filter.class_parameters !== undefined &&
-            incoming.filter.class_parameters["LASTHEADER"] !== undefined
-          ) {
-            incoming.filter.class_parameters["LASTHEADER"] = id;
+          let id = undefined;
+          try {
```

Comment:
Set id in .then section and  reject in .catch section. TryCatch block is not needed, just await sendDataTo.Grid(incoming.descr).then(e){...}.catch(e){reject(e)}

Reply:
Fixed. I reorganized this part of the code for the better understanding of the process, including that this try/catch mechanism was promisified. Please validate if it is the correct way! Thank you for the feedback!